src/system_simulation/trace_visualization.rb

```diff
@@ -0,0 +1,126 @@
+# Simulate data samples of a system simulation by plotting a trace consisting of transparent circles into the scene.
+class TraceVisualization
+  def initialize
+    # Simulation data to visualize
+    @simulation_data = nil
+
+    # Group containing trace circles.
+    @group = Sketchup.active_model.active_entities.add_group
+
+    # List of trace circles.
+    @trace_points = []
+
+    # Visualization parameters
+    @color = Sketchup::Color.new(72,209,204)
+    @alpha = 0.4
+  end
+
+  def add_trace(node_ids, sparce_factor, data)
+    @simulation_data = data
+    add_circle_trace(node_ids, sparce_factor)
+  end
+
+  def reset_trace()
+    @group = Sketchup.active_model.entities.add_group if @group.deleted?
+    Sketchup.active_model.active_entities.erase_entities(@group.entities.to_a)
+    if @trace_points.count > 0
+      Sketchup.active_model.active_entities.erase_entities(@trace_points)
+    end
+    @trace_points = []
+  end
+
+  private
+
+  def add_circle_trace(node_ids, sparse_factor)
```

Comment:
it seems we only use node_ids[0] and node_ids[1] here, but I don't get if there are more node_ids in there. Otherwise I would propse to make the add_circle_trace method only take one node_id, and then call that two times.

Reply:
Fixed in 60534d5